app/controllers/user/shares_controller.rb

```diff
@@ -29,52 +29,46 @@ def new
   # POST /shares
   def create
     require_valid_params
-
-    @share = create_share_with_auditing
-
-    if @share.valid?
-      respond_with(@share, location: user_birth_record_path(@share.document))
-    else
-      respond_with(@share)
-    end
+    @share = Share.new(recipient: find_organisation, document: find_document, user: current_user)
+    authorize @share
+    @share.save
+    respond_with(@share, location: user_document_path(@share.document.document_type, @share.document.id))
   end
 
   def revoke
     @share.revoke(revoked_by: current_user)
-    respond_with(@share, location: user_birth_record_path(@share.document))
+    respond_with(@share, location: user_document_path(@share.document.document_type, @share.document.id))
   end
 
   private
 
-  def create_share_with_auditing
-    birth_record = current_user.birth_records.find_by(id: share_params['document_id'])
-    authorize birth_record, :share?
-    recipient = Organisation.find_by(id: share_params['recipient_id'])
-    birth_record.share_with(
-      recipient: recipient,
-      user: current_user
-    )
-  end
-
   # Set the share, if it exists and is available to the current user
   def set_share
     @share = user_shares.find_by(params.permit(:id))
     authorize @share
   end
 
+  # Never trust parameters from the scary internet, only allow the white list through.
+  def share_params
+    params.require(:share).permit(:document_id, :document_type, :recipient_id)
+  end
+
   # Require the params which will allow a valid model to be created
   #
   # The return value isn't useful, but this will raise an error if parameters
   # are missing
   def require_valid_params
     params
       .require(:share)
-      .require([:document_id, :recipient_id])
+      .require(%i[document_id document_type recipient_id])
```

Comment:
why do we need this method? Won't the model validators and pundit permissions do this for us?